src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1225,6 +1225,87 @@ module T = struct
   let dummy_transaction_pool_proxy : transaction_pool_proxy =
     { find_by_hash = const None }
 
+  let extract_zkapp_veirfication_keys
+      ({ account_updates; _ } : Zkapp_command.Verifiable.t) : _ Result.t =
```

Comment:
I just realized that we can simplify the flow. Instead of comparing the verification keys, let's do the equality check right in this function (and it'll return `Ok ()` if everything is ok).

See my suggestions below.